src/components/DocumentStatus/DocumentStatus.tsx

```diff
@@ -10,19 +10,39 @@ interface DocumentStatusProps {
   className?: string;
 }
 
+interface VerificationFragmentData {
+  did: string;
+  location: string;
+  status: string;
+}
+
 export const IssuedBy = ({ verificationStatus }: DocumentStatusProps) => {
-  const dnsFragmentName = "OpenAttestationDnsTxt";
-  const dnsFragment = verificationStatus.find((status) => status.name === dnsFragmentName);
-  const dnsIdentity = dnsFragment?.data?.every((issuer: { status: string }) => issuer.status === "VALID");
-  const domainNames = dnsFragment?.data
-    ?.map((issuer: { location: string }) => issuer.location.toUpperCase())
-    .join(", ");
-  const formattedDomainNames = domainNames?.replace(/,(?=[^,]*$)/, " and"); // regex to find last comma, replace with and
+  const formatDomainNames = (fragment: VerificationFragment<VerificationFragmentData[]>): string | undefined => {
```

Comment:
```suggestion
  const formatIdentifier = (fragment: VerificationFragment<VerificationFragmentData[]>): string | undefined => {
```

It's no longer just domain names